src/client.rs

```diff
@@ -653,16 +711,35 @@ impl Client {
     }
 }
 
-#[cfg(feature = "in-use-encryption-unstable")]
 #[derive(Clone, Debug)]
 pub(crate) struct WeakClient {
-    inner: std::sync::Weak<ClientInner>,
+    inner: crate::tracking_arc::Weak<ClientInner>,
 }
 
-#[cfg(feature = "in-use-encryption-unstable")]
 impl WeakClient {
-    #[allow(dead_code)]
     pub(crate) fn upgrade(&self) -> Option<Client> {
         self.inner.upgrade().map(|inner| Client { inner })
     }
 }
+
+#[derive(Derivative)]
+#[derivative(Debug)]
+pub(crate) struct AsyncDropToken {
+    #[derivative(Debug = "ignore")]
+    tx: Option<tokio::sync::oneshot::Sender<BoxFuture<'static, ()>>>,
+}
+
+impl AsyncDropToken {
+    pub(crate) fn spawn(&mut self, fut: impl Future<Output = ()> + Send + 'static) {
```

Comment:
Ideally this would consume the token, but since `drop` only gives a `&mut self` this follows suit and uses an internal `Option`.